.github/workflows/ci.yml

```diff
@@ -292,12 +292,17 @@ jobs:
       - name: "Extracting build dir archive"
         run: tar -xf "${BUILD_DIR_ARCHIVE}"
       - name: "Testing all"
+        # Note: VK_LOADER_LAYERS_ENABLE is used to disable the
+        # Vulkan Validation Layers (VVL), since we have unresolved errors on
+        # SwiftShader.
         run: |
           ./build_tools/github_actions/docker_run.sh \
             --env IREE_CUDA_DISABLE=1 \
+            --env VK_LOADER_LAYERS_ENABLE= \
```

Comment:
Setting `VK_LOADER_LAYERS_ENABLE` to an empty list works on my local machine to disable the validation layers, and our CI has a recent enough Vulkan loader to use that environment variable (1.3.234+ according to https://vulkan.lunarg.com/doc/view/latest/windows/layer_configuration.html). This is still using the layer though ([logs](https://github.com/openxla/iree/actions/runs/7226920714/job/19693780041?pr=15884)).

Otherwise, I was thinking about ways to pass IREE's `--vulkan_validation_layers` flag to tests or change the value of that flag based on a new environment variable.

I think what I want to end up with in CI builds is:

* SwiftShader with validation layers enabled (while SwiftShader is still in use) OR SwiftShader without validation layers (if we want to punt on getting validation-clean)
* Desktop NVIDIA with validation layers enabled
* Desktop AMD with validation layers enabled
* Desktop NVIDIA and/or AMD with a min-spec profile, validation layers enabled
* Android Mali with validation layers enabled
* etc. (new builds could choose how much checking they want)

In order to get there, we can either fix all validation layer errors and then switch over in one go, or we can make incremental progress by configuring the "enable the validation layers" and "check for validation layer errors" flags for each CI job.

These CI workflows could run ctest multiple times (in the same step, in multiple steps, or in a matrix), with each run using a different set of environment variables (see [these docs](https://vulkan.lunarg.com/doc/view/latest/windows/profiles_layer.html)) to configure Vulkan.